phys/module_pbl_driver.F

```diff
@@ -2260,6 +2269,25 @@ SUBROUTINE pbl_driver(                                          &
               ,ITS=its,ITE=ite,JTS=jts,JTE=jte,KTS=kts,KTE=kte)
           ENDIF
 
+     IF (distributed_ahe_opt > 0) THEN
+       call cal_mon_day(julday, julyr, jmonth, jday)
+       ihour = (jmonth - 1) * 24 + MOD(INT(gmt + xtime / 60.0), 24)
+       IF (distributed_ahe_opt == 1) THEN
+         DO j = jts, jte
+         DO i = its, ite
+           ! Volumetric heat capacity of air = 1200 J/(K m3)
+           RTHBLTEN(i, 1, j) = RTHBLTEN(i, 1, j) + ahe(i, ihour, j) / 1200 / DZ8W(i, 1, j)
+         END DO
+         END DO
+       ELSE IF (distributed_ahe_opt == 2) THEN
+         DO j = jts, jte
+         DO i = its, ite
+           HFX(i, j) = HFX(i, j) + ahe(i, ihour, j)
+         END DO
+         END DO
+       END IF
+     END IF
+
```

Comment:
@epn09 @cenlinhe Why is HFX added at the end of PBL driver? What is the added HFX for?

Reply:
this is for anthropogenic heat flux (i.e., sensible heat flux) released from surface human activities to the atmosphere.

Reply:
@cenlinhe After PBL call, how do you expect the flux to be affecting the atmosphere?

Reply:
I think @epn09 originally wants to let this flux affect atmosphere for the next time step. Another technical consideration is that @epn09 wants to add this flux to the temperature tendency (ahe_option=2) which is an output of PBL scheme, so this part needs to be added at the end of PBL call. This is my understanding if I remember his earlier explanations.

Reply:
@weiwangncar @cenlinhe 
* I think we added anthropogenic heat (AH) to the end of PBL so that it will not be overwritten by any other scheme.
* Already discussed before but AH is added in PBL instead of UCM because AH can present in non-urban grids.
* There are two options for AH:
   1. Add to the temperature tendency at the first layer of the atmosphere
   2. Add directly to the surface sensible heat flux.
* So far, we found that the first option often produce more natural (?) result but we'd like to expose both options to WRF users.

Reply:
@weiwangncar I see. So where would be the right place to add AH to HFX?

Reply:
@epn09 Perhaps you could do this in module_pbl_driver.F, but before any PBL physics is called.

Reply:
Some PBL schemes may set RTHBLTEN to zero before adding to it. It needs to be checked.